src/figanos/hvplot/plot.py

```diff
@@ -0,0 +1,374 @@
+            name,
+            arr,
+            array_categ,
+            array_data,
+            plot_kw,
+            opts_kw,
+            legend,
+            form,
+            use_attrs,
+        )
+        #  non-ensemble Datasets
+    elif array_categ[name] == "DS":
+        ignore_label = False
+        for k, sub_arr in arr.data_vars.items():
+            sub_name = (
+                sub_arr.name if non_dict_data is True else (name + "_" + sub_arr.name)
+            )
+            #  if kwargs are specified by user, all lines are the same and we want one legend entry
+            if plot_kw[name]:
+                label = name if not ignore_label else ""
+                ignore_label = True
+            else:
+                label = sub_name
+            hv_fig[sub_name] = sub_arr.hvplot.line(
+                x="time", label=label, **plot_kw[name]
+            ).opts(**opts_kw[name])
+
+        #  non-ensemble DataArrays
+    elif array_categ[name] in ["DA"]:
+        return arr.hvplot.line(label=name, **plot_kw[name]).opts(**opts_kw[name])
+    else:
+        raise ValueError(
+            "Data structure not supported"
+        )  # can probably be removed along with elif logic above,
+        # given that get_array_categ() also does this check
+    if hv_fig:
+        return hv_fig
+
+
+def timeseries(
+    data: Union[dict[str, Any], xr.DataArray, xr.Dataset],
+    use_attrs: Union[dict[str, Any], None] = None,
+    plot_kw: Union[dict[str, Any], None] = None,
+    opts_kw: Union[dict[str, Any], None] = None,
+    legend: str = "lines",
+    show_lat_lon: Union[bool, str, int, tuple[float, float]] = True,
+) -> Union[hv.element.chart.Curve, hv.core.overlay.Overlay]:
+    """Plot time series from 1D Xarray Datasets or DataArrays as line plots.
+
+    Parameters
+    ----------
+    data : dict or Dataset/DataArray
+        Input data to plot. It can be a DataArray, Dataset or a dictionary of DataArrays and/or Datasets.
+    use_attrs : dict, optional
+        A dict linking a plot element (key, e.g. 'title') to a DataArray attribute (value, e.g. 'Description').
+        Default value is {'title': 'description', 'ylabel': 'long_name', 'yunits': 'units'}.
+        Only the keys found in the default dict can be used.
+    plot_kw : dict, optional
+        Arguments to pass to the `hvplot.line()` or hvplot.area() function. Changes how the line looks.
+        If 'data' is a dictionary, must be a nested dictionary with the same keys as 'data'.
+    opts_kw: dict, optional
+        Arguments to pass to the `holoviews/hvplot.opts()` function. Changes figure options and access to hooks.
+        If 'data' is a dictionary, must be a nested dictionary with the same keys as 'data' to pass nested to each
+        individual figure or key 'overlay' to pass to overlayed figures.
+    legend : str (default 'lines') or dict
+        'full' (lines and shading), 'lines' (lines only), 'in_plot' (end of lines), 'none' (no legend).
```

Comment:
would it be possible to have "edge" like matplotlib?

Reply:
I coded a similar function to edge and it didn't look good since it couldn't follow the plotted lines when you zoomed in or out of the figure. It made more sense to me to remove that option for interactive figures

Reply:
ok!